cluster/service/cache.go

```diff
@@ -160,6 +160,7 @@ func (c *cache) createClient(ctx context.Context) (*client.Client, error) {
 	}
 
 	httpClient := http.DefaultClient
+	httpClient.Timeout = 10 * time.Second
```

Comment:
that would actually let the `auth` service wait for 10s until the `cluster` service is up, but what about using an env var for this, so we can increase the timeout from the config maps (we could have a default value of 10s in the configuration)

Reply:
Well.. the first attempt of getting the list of the clusters happens during startup in main(). And this attempt is not critical. If it fails then we move on and will try to get the list again when we really need it or every 5 minutes by the cluster cache refresher whichever occurs first. It's done in that way to avoid cycle dependencies between auth and cluster during startup.
 
The first main() attempt is what we would like to make as short as possible. But the cache refresh and lazy initialization things is more critical and I would keep them 30 seconds as we have now.

So, my suggestion is to use `options` parameter in `Start()` function - https://github.com/fabric8-services/fabric8-auth/blob/master/cluster/service/cluster.go#L60 when called from `main()` - https://github.com/fabric8-services/fabric8-auth/blob/master/main.go#L176

Something like this:
```
err = clusterservice.Start(context.Background(), config, func(c *http.Client){
	c.Timeout = 3 * time.Second
})
```
I would keep the timeout in main() very short. 3 or maybe 5 seconds, not longer. And would keep the rest of the code intact.

Reply:
you are right. Now using `options` with Timeout of 3s for first attempt from main